subprojects/core/src/main/java/org/gradle/execution/plan/Node.java

```diff
@@ -37,6 +37,22 @@
 public abstract class Node implements Comparable<Node> {
     private static final Logger LOGGER = LoggerFactory.getLogger(Node.class);
 
+    /**
+     * Notifies the work properties of the start of execution, so they may finalize and cache whatever state is required to efficiently fingerprint inputs and outputs, apply validation or whatever.
```

Comment:
"or whatever" sounds a bit weird.

Also, I think we should add something about how this makes the values practically immutable from that point on.